samples/__tests__/e2e.test.ts

```diff
@@ -159,22 +159,42 @@ class PollingInboundTransporter implements InboundTransporter {
 }
 
 class HttpOutboundTransporter implements OutboundTransporter {
+  private agent: Agent
+
+  public constructor(agent: Agent) {
+    this.agent = agent
+  }
   public async sendMessage(outboundPackage: OutboundPackage, receiveReply: boolean) {
     const { payload, endpoint } = outboundPackage
 
     if (!endpoint) {
       throw new Error(`Missing endpoint. I don't know how and where to send the message.`)
     }
 
-    testLogger.test(`Sending outbound message to connection ${outboundPackage.connection.id}`, outboundPackage.payload)
+    logger.test(`Sending outbound message to connection ${outboundPackage.connection.id}`, outboundPackage.payload)
 
     if (receiveReply) {
       const response = await post(`${endpoint}`, JSON.stringify(payload))
+      logger.test(`received response:\n ${response}`)
       const wireMessage = JSON.parse(response)
-      testLogger.test('received response', wireMessage)
-      return wireMessage
+      this.agent.receiveMessage(wireMessage)
```

Comment:
We had this wrong before you made the changes, but as you're already updating this code. receiveReply means we MAY receive a reply. We're currently just parsing it without checking if we received a reply.

Reply:
Yeah, good point. Updated.